packages/legend-application-studio/src/stores/graphModifier/DomainGraphModifierHelper.ts

```diff
@@ -90,6 +91,54 @@ export const class_addProperty = action(
   },
 );
 
+export const class_arrangeProperty = action(
+  (_class: Class, sourceProperty: Property, targetProperty: Property): void => {
+    const sourceIndex = _class.properties.findIndex(
+      (property) => property === sourceProperty,
+    );
+    const targetIndex = _class.properties.findIndex(
+      (property) => property === targetProperty,
+    );
+
+    if (sourceIndex < targetIndex) {
+      const tempProperty = targetProperty;
+      changeEntry(_class.properties, targetProperty, sourceProperty);
+      changeEntry(_class.properties, sourceProperty, tempProperty);
+    } else {
+      const tempProperty = sourceProperty;
+
+      changeEntry(_class.properties, sourceProperty, targetProperty);
+      changeEntry(_class.properties, targetProperty, tempProperty);
+    }
+  },
+);
+
+export const class_arrangeDerivedProperty = action(
+  (
+    _class: Class,
+    sourceProperty: DerivedProperty,
+    targetProperty: DerivedProperty,
+  ): void => {
+    const sourceIndex = _class.derivedProperties.findIndex(
+      (property) => property === sourceProperty,
+    );
+    const targetIndex = _class.derivedProperties.findIndex(
+      (property) => property === targetProperty,
+    );
+
+    if (sourceIndex < targetIndex) {
+      const tempProperty = targetProperty;
```

Comment:
Strange though, do we really need to have this `temp` here?

Reply:
The temp is needed for this way, or rather when writing this I was trying to implement rearranging but with changeEntry it replies old entry with new entry so when I solely had changeEntry(source, target) I would lose information. No swap entry in the common utils so this was sort of a workaround I did